Change dust control

lontivero opened the pull request:
This PR is for changing the way Wasabi deals with dust coins. Currently Wasabi keeps the dust coins but with a mark that can be queried to know when it is relevant or not. With this PR dust coins are completely ignored by Wasabi as if they have never existed.

Close #1383

nopara73 commented:
NACK for now, because dust protection works for me. We have to figure out what's the issue.

nopara73 commented:
As I said here: https://github.com/zkSNACKs/WalletWasabi/issues/1383#issuecomment-487538918

Dust protection doesn't work after restart. So not NACK.

nopara73 commented:
NACK for this PR. This is a hack to hide things in the UI. The whole reason we are in this situation is, because I was hacking in the first place. Now you want to hack around my hacking? That won't end well.

lontivero commented:
I don't understand. How is this a hack? I don't what to hack anything here, I've just removed all the flags everywhere, removed many lines of code and replace it for 2 lines. 

There is something that I don't get here: you say  that "_Dust protection doesn't work after restart_". Are you talking about my change or about the current situation (before this PR)?

nopara73 commented:
Hey Lucas, just a heads up that I didn't forget this PR. I'll either merge or rework this before v1.1.5 is released.

lontivero commented:
Good to know because I the `dustThreshold` is `0.0001` and I have dust in my testnet wallet:
![image](https://user-images.githubusercontent.com/127973/58275219-e6946100-7d6a-11e9-8335-e5b78921d41e.png)

nopara73 commented:
1. You removed the GUI checks.
2. You removed the `IsDust` `SmartCoin` property and all its checks.
3. You made sure the `ProcessTransaction` doesn't concern itself with outputs those are <= dust.

Indeed it's not a hack, but a very clean solution. Why didn't I implement it this way in the first place? I'm struggled. I was either stupid or I had a very good reason I don't remember.
